druid/src/main/java/org/apache/calcite/adapter/druid/DruidRules.java

```diff
@@ -113,20 +113,36 @@ private DruidRules() {}
           SORT_PROJECT_TRANSPOSE);
 
   /** Predicate that returns whether Druid can not handle an aggregate. */
-  private static final Predicate<Aggregate> BAD_AGG =
-      new PredicateImpl<Aggregate>() {
-        public boolean test(Aggregate aggregate) {
+  private static final Predicate<Pair<Aggregate, DruidTable>> BAD_AGG =
+      new PredicateImpl<Pair<Aggregate, DruidTable>>() {
+        public boolean test(Pair<Aggregate, DruidTable> pair) {
+          final Aggregate aggregate = pair.left;
+          final DruidTable druidTable = pair.right;
           final CalciteConnectionConfig config =
                   aggregate.getCluster().getPlanner().getContext()
                       .unwrap(CalciteConnectionConfig.class);
+          RelNode input = aggregate.getInput();
           for (AggregateCall aggregateCall : aggregate.getAggCallList()) {
             switch (aggregateCall.getAggregation().getKind()) {
             case COUNT:
-              if (!aggregateCall.getArgList().isEmpty()) {
-                // Cannot handle this aggregate function
+              // Druid can handle 2 scenarios:
+              // 1. count(distinct col) when approximate results
+              //    are acceptable and col is not a metric
+              // 2. count(*)
+
+              // Make sure no column name is a metric
+              for (int index : aggregateCall.getArgList()) {
+                String name = input.getRowType().getFieldNames().get(index);
```

Comment:
Does the query

```sql
select "B", count(distinct "A") from (select "unit_sales" as "A", "customer_id" as "B" from "foodmart") group by "B";
```

Satisfy your above comment for an additional test? (I'll also add a count distinct variant)